src/client.rs

```diff
@@ -0,0 +1,129 @@
+#![deny(missing_docs)]
```

Comment:
Since you put `#![deny(missing_docs)]` in every file, you can just put it in lib. This should apply it across the entire project.

Reply:
Nice, I wasn't aware it applied it to all submodules when used at a higher module level.